### PR TITLE
Chore/circleci cross build

### PR DESCRIPTION
**Pull request by sky1045**
build launcher artifact for every OS(windows, macOS, linux) from debian docker container.
- Didn't remove run-script-os considering local environment
- cache headless per os not arch
- upgrade electron-packager for apple sillicon
- upgrade electron-log for windows

### CODE REVIEW

**Comment by longfin** [comment]
@BasixKOR Could you check [it](https://app.circleci.com/pipelines/github/planetarium/9c-launcher/1776/workflows/7e38a3b9-943e-41c5-b6ec-6c404e7fc322/jobs/7140?invite=true#step-112-539)?

**Comment by BasixKOR** [comment]
I've merged the latest commit to using Playwright instead of Spectron. I think it should resolve the ChromeDriver issue.

File: .circleci/config.yml [hunk 1]
```diff
@@ -67,100 +58,81 @@ commands:
...
     parameters:
       os:
-        type: executor
-    executor: << parameters.os >>
+        type: string
     steps:
       - when:
           condition:
-            equal: [*windows, << parameters.os >>]
+            equal: [ windows, << parameters.os >> ]
```
Comment (BasixKOR) [inline .circleci/config.yml:73]:
```suggestion
            equal: [windows, << parameters.os >>]
```

Just bikeshedding formatting.

File: package.json [hunk 2]
```diff
@@ -210,7 +206,7 @@
     "electron-devtools-installer": "^3.2.0",
     "electron-download": "^4.1.1",
     "electron-mock-ipc": "^0.3.12",
-    "electron-packager": "^14.2.1",
+    "electron-packager": "15.3.0",
```
Comment (BasixKOR) [inline package.json:209]:
```suggestion
    "electron-packager": "^15.3.0",
```

This allows us to update to a newer version when there are no breaking changes, per the [Semantic Versioning](https://semver.org/).

File: .circleci/config.yml [hunk 3]
```diff
@@ -67,100 +58,81 @@ commands:
...
             - run:
                 command: |
-                  curl "https://awscli.amazonaws.com/AWSCLIV2.pkg" -o "AWSCLIV2.pkg"
-                  sudo installer -pkg AWSCLIV2.pkg -target /
-                name: Install AWS CLI
+                  echo "export OS_ALIAS=darwin" >> $BASH_ENV
+                  echo "export ARTIFACT=MacOS.tar.gz" >> $BASH_ENV
+                  echo "export EXECUTABLE='Nine Chronicles.app/Contents/Resources/app/9c.app/Contents/MacOS/9c'" >> $BASH_ENV
+                  echo "export ZIP_CMD='tar cvfz'" >> $BASH_ENV
+                name: Setup Mac OS Variables
```
Comment (BasixKOR) [inline .circleci/config.yml:103]: That's awesome trick! You might want to split this off to another script like `configure_artifact.sh` with conditionals.

```bash
if [ $0 = "macos" ]; then
	echo "export OS_ALIAS=darwin" >> $BASH_ENV
  echo "export ARTIFACT=MacOS.tar.gz" >> $BASH_ENV
  echo "export EXECUTABLE='Nine Chronicles.app/Contents/Resources/app/9c.app/Contents/MacOS/9c'" >> $BASH_ENV
  echo "export ZIP_CMD='tar cvfz'" >> $BASH_ENV
else
  echo "export OS_ALIAS=linux" >> $BASH_ENV
  echo "export ARTIFACT=Linux.tar.gz" >> $BASH_ENV
  echo "export EXECUTABLE=resources/app/9c" >> $BASH_ENV
  echo "export ZIP_CMD='tar cvfz'" >> $BASH_ENV
fi
```

File: .circleci/config.yml (hunk 1 above)
Comment (BasixKOR) [reply .circleci/config.yml:73]: Probably you want to run the Prettier once on the file, like `yarn prettier --write .circleci/config.yml`

**Review by BasixKOR — APPROVED** [review]
Looks good to me.

**Pull request merged**